distribution/xfer/download.go

```diff
@@ -121,20 +121,20 @@ func (ldm *LayerDownloadManager) Download(ctx context.Context, initialRootFS ima
 			if err == nil {
 				getRootFS := rootFS
 				getRootFS.Append(diffID)
-				l, err := ldm.layerStores[string(os)].Get(getRootFS.ChainID())
+				l, err := ldm.layerStores[os].Get(getRootFS.ChainID())
```

Comment:
Again, `rootfs.Layer()` would avoid this.

Reply:
For the rest, they could be also avoided with some better api that is mapped to a specific layerstore. But that can be added later. For now, I think I'd prefer just passing the `layerStore` instance here rather than the `os` strings.

Reply:
@tonistiigi Can you clarify what you mean by your 'For now...' bit? I can't see what you're suggesting. Thx.